### ISSUE TITLE
Use the new builtin `nvim_set_keymap()` `desc` option for WhichKey integration

### ISSUE THREAD

**Issue by b0o** [b0o/mapx.nvim#14]
This would dramatically simplify the code by obviating the need for direct coupling with WhichKey.

This can be partially implemented immediately by passing the `desc` option to `nvim_set_keymap()`, but the removal of WhichKey-specific code depends on https://github.com/folke/which-key.nvim/pull/253 / https://github.com/folke/which-key.nvim/issues/242.

**Comment by towry** [comment]
Both upstream dependencies seem resolved, tag `blocked` can be removed ?

**Comment by b0o** [comment]
Yes, thanks for reminding me @towry – I don’t have time to work on this right now but I’m open to PRs.
